services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/permissions/HankeKayttajaControllerITest.kt

```diff
@@ -948,6 +976,179 @@ class HankeKayttajaControllerITest(@Autowired override val mockMvc: MockMvc) : C
             }
         }
     }
+
+    @Nested
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    inner class CheckForDelete {
+        private val kayttajaId = UUID.fromString("294af703-7982-47f8-bc08-f08ace485a2b")
+        private val url = "/kayttajat/$kayttajaId/deleteInfo"
+        private val hankeId = 34597
+        private val otherKayttajaId = UUID.fromString("77f896ee-2c2a-4727-bb73-c04ddd30f3b2")
+
+        @Test
+        fun `Returns not found when the call is not authorized`() {
+            every { authorizer.authorizeKayttajaId(kayttajaId, DELETE_USER.name) } throws
+                HankeKayttajaNotFoundException(kayttajaId)
+
+            get(url).andExpect(status().isNotFound).andExpect(hankeError(HankeError.HAI4001))
+
+            verifySequence { authorizer.authorizeKayttajaId(kayttajaId, DELETE_USER.name) }
+        }
+
+        @Test
+        fun `Returns not found when hankekayttaja is not found`() {
+            every { authorizer.authorizeKayttajaId(kayttajaId, DELETE_USER.name) } returns true
+            every { hankeKayttajaService.getKayttaja(kayttajaId) } throws
+                HankeKayttajaNotFoundException(kayttajaId)
+
+            get(url).andExpect(status().isNotFound).andExpect(hankeError(HankeError.HAI4001))
+
+            verifySequence {
+                authorizer.authorizeKayttajaId(kayttajaId, DELETE_USER.name)
+                hankeKayttajaService.getKayttaja(kayttajaId)
+            }
+        }
+
+        @Test
+        fun `Returns true for onlyOmistajanYhteyshenkilo when the user is only contact for omistaja`() {
+            every { authorizer.authorizeKayttajaId(kayttajaId, DELETE_USER.name) } returns true
+            every { hankeKayttajaService.getKayttaja(kayttajaId) } returns
+                HankeKayttajaFactory.create(id = kayttajaId, hankeId = hankeId)
+            every { hankeService.loadHankeById(hankeId) } returns
+                HankeFactory.create(id = hankeId)
+                    .withOmistaja(1, 1, HankeYhteyshenkiloFactory.create(id = kayttajaId))
+            every { hakemusService.getHakemuksetForKayttaja(kayttajaId) } returns listOf()
+
+            val response: DeleteInfo = get(url).andExpect(status().isOk).andReturnBody()
+
+            assertThat(response).all {
+                prop(DeleteInfo::onlyOmistajanYhteyshenkilo).isTrue()
+                prop(DeleteInfo::activeHakemukset).isEmpty()
```

Comment:
In most of these tests only the thing mentioned in the test name is checked. E.g. onlyOmistajanYhteyshenkilo. Here, also activeHakemukset and draftHakemukset are checked. Should they be checked in every test or removed from here?

Reply:
I added these here so I wouldn't have to make a new test case that checks these are empty when there are no applications for the user. Integration tests are not free in running time, after all. Would it be better if that was a separate test?

Reply:
I think this is fine as it is now.